src/avram/operation.cr

```diff
@@ -1,23 +1,82 @@
 require "./validations"
+require "./callbacks"
 require "./define_attribute"
 require "./save_operation_errors"
 require "./param_key_override"
+require "./needy_initializer"
 
-class Avram::VirtualForm
-  macro inherited
-    {% raise "Avram::VirtualForm has been renamed to Avram::Operation. Please inherit from Avram::Operation." %}
-  end
-end
-
-class Avram::Operation
+abstract class Avram::Operation
+  include Avram::NeedyInitializer
   include Avram::DefineAttribute
   include Avram::Validations
   include Avram::SaveOperationErrors
```

Comment:
This isn't really `SaveOperation` specific. I could rename it, or that could come in a future PR I'm planning with some cleanup stuff.

Reply:
Future PR sounds good! `Avram::OperationsErrors` seems like a good name